### ISSUE TITLE
[release] test failed due to app config building error

### ISSUE THREAD

**Issue by iycheng** [ray-project/ray#20503]
stress_test_many_tasks/many_actor_tests/node_failures failed due to app config building error.

Some example are here:
https://buildkite.com/ray-project/periodic-ci/builds/1700#15650bd3-b052-45fc-aa98-13c3a3cbbb4f

https://beta.anyscale.com/o/anyscale-internal/configurations/app-config-details/bld_SXWhBBrSXaYp24EVwSx1q76X

**Comment by zhe-thoughts** [comment]
Thanks for catching this @iycheng . Why did this test pass for RC0 and failing now?

**Comment by iycheng** [comment]
I'm not quite sure about this. It seems pip change. @wuisawesome  do you have some ideas?

**Comment by wuisawesome** [comment]
It seems to be failing because doing 

```
pip install -U ray@<url>
pip install ray[all]
```

the second line is downgrading the ray version. That isn't the behavior I'd expect from pip though. We can work around it pretty easily, but still not sure why it behaves like that.

**Comment by wuisawesome** [comment]
@iycheng I just realized that these logs come from 2 separate tests (both should be release blockers)

many_tasks fails on the app config build: https://beta.anyscale.com/o/anyscale-internal/configurations/app-config-details/bld_SXWhBBrSXaYp24EVwSx1q76X

stress_test_many_tasks failed for a different reason. Looks like some edge case that `e2e.py` just didn't handle around app config caching? I retried that job and now it's running: https://console.anyscale.com/o/anyscale-internal/projects/prj_2xR6uT6t7jJuu1aCwWMsle/clusters/ses_rDz9Bd37Zsfsy4pWwhYTSw7e

As for why `many_tasks` failed, it looks like this 

```
pip install ray@<url>
pip install ray
```

is downgrading Ray. I'm pretty sure this isn't the typical pip behavior (it usually doesn't upgrade/downgrade versions without `--ugprade` being passed...). There's probably some edge case around a dependency pinning the ray version or something, but regardless, we can easily work around it.
